tests/components/blebox/test_light.py

```diff
@@ -0,0 +1,597 @@
+
+    state = hass.states.get(entity_id)
+    assert state.state == STATE_OFF
+
+    hs_color = color.color_RGB_to_hs(0xFF, 0xA1, 0xB2)
+
+    def turn_on(value):
+        feature_mock.is_on = True
+        assert value == "ffa1b2e4"
+        feature_mock.white_value = 0xE4
+        feature_mock.rgbw_hex = value
+
+    feature_mock.async_on = AsyncMock(side_effect=turn_on)
+
+    def apply_color(value, color_value):
+        assert value == "c1a2e3e4"
+        assert color_value == "ffa0b1"
+        return "ffa1b2e4"
+
+    feature_mock.apply_color = apply_color
+    feature_mock.sensible_on_value = "c1a2e3e4"
+
+    await hass.services.async_call(
+        "light",
+        SERVICE_TURN_ON,
+        {"entity_id": entity_id, ATTR_HS_COLOR: hs_color},
+        blocking=True,
+    )
+
+    state = hass.states.get(entity_id)
+    assert state.attributes[ATTR_HS_COLOR] == hs_color
+    assert state.attributes[ATTR_WHITE_VALUE] == 0xE4
+    assert state.state == STATE_ON
+
+
+async def test_wlightbox_on_to_last_color(wlightbox, hass, config):
+    """Test light on."""
+
+    feature_mock, entity_id = wlightbox
+
+    def initial_update():
+        feature_mock.is_on = False
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    await async_setup_entity(hass, config, entity_id)
+    feature_mock.async_update = AsyncMock()
+
+    state = hass.states.get(entity_id)
+    assert state.state == STATE_OFF
+
+    def turn_on(value):
+        feature_mock.is_on = True
+        assert value == "f1e2d3e4"
+        feature_mock.white_value = 0xE4
+        feature_mock.rgbw_hex = value
+
+    feature_mock.async_on = AsyncMock(side_effect=turn_on)
+    feature_mock.sensible_on_value = "f1e2d3e4"
+
+    await hass.services.async_call(
+        "light", SERVICE_TURN_ON, {"entity_id": entity_id}, blocking=True,
+    )
+
+    state = hass.states.get(entity_id)
+    assert state.attributes[ATTR_WHITE_VALUE] == 0xE4
+    assert state.attributes[ATTR_HS_COLOR] == color.color_RGB_to_hs(0xF1, 0xE2, 0xD3)
+    assert state.state == STATE_ON
+
+
+async def test_wlightbox_off(wlightbox, hass, config):
+    """Test light off."""
+
+    feature_mock, entity_id = wlightbox
+
+    def initial_update():
+        feature_mock.is_on = True
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    await async_setup_entity(hass, config, entity_id)
+    feature_mock.async_update = AsyncMock()
+
+    state = hass.states.get(entity_id)
+    assert state.state == STATE_ON
+
+    def turn_off():
+        feature_mock.is_on = False
+        feature_mock.white_value = 0x0
+        feature_mock.rgbw_hex = "00000000"
+
+    feature_mock.async_off = AsyncMock(side_effect=turn_off)
+
+    await hass.services.async_call(
+        "light", SERVICE_TURN_OFF, {"entity_id": entity_id}, blocking=True,
+    )
+
+    state = hass.states.get(entity_id)
+    assert ATTR_WHITE_VALUE not in state.attributes
+    assert ATTR_HS_COLOR not in state.attributes
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5535891)